dev/src/validate.ts

```diff
@@ -14,194 +14,313 @@
  * limitations under the License.
  */
 
-import * as is from 'is';
-
 import {FieldPath} from './path';
-import {AnyDuringMigration} from './types';
+import {isFunction, isObject} from './util';
 
 /**
- * Formats the given word as plural conditionally given the preceding number.
+ * Options to allow argument omission.
  *
  * @private
  */
-function formatPlural(num, str) {
-  return `${num} ${str}` + (num === 1 ? '' : 's');
+export interface AllowOptional {
```

Comment:
nit: I don't love `AllowOptional` and `AllowRange` as names.  Normally I like to name classes / interfaces as nouns.  I think maybe `OptionalOption` and `RangeOptions` would be better, though admittedly `OptionalOption` is a bit awkward. :)